### PR TITLE
.Net: Add model documentation

### PR DESCRIPTION
**Pull request by matthewbolanos**
To provide guidance to the community, we'll be creating a formal list of models we would like to support within the core repo of Semantic Kernel.

### CODE REVIEW

File: dotnet/docs/MODELS.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+| P0       | microsoft-phi-2   | Not started | `ITextGeneration`      | Azure AI        | TBD          | TBD       | TBD      |
+| P0       | microsoft-phi-2   | Not started | `ITextGeneration`      | Hugging Face    | TBD          | TBD       | TBD      |
+| P2       | resnet-50         | Not started | `IImageClassification` | Azure AI        | TBD          | TBD       | TBD      |
+| P2       | resnet-50         | Not started | `IImageClassification` | Hugging Face    | TBD          | TBD       | TBD      |
+
+### Google
+
+| Priority | Model             | Status      | Interface              | Deployment type | GitHub issue | Developer | Reviewer |
+| -------- | ----------------- | ----------- | ---------------------- | --------------- | ------------ | --------- | -------- |
+| P0       | gemini-pro        | Not started | `IChatCompletion`      | Google API      | TBD          | TBD       | TBD      |
```
Comment (Krzysztof318) [inline dotnet/docs/MODELS.md:61]: Change to in-progress. Add also VertexAI deployment type.

File: dotnet/docs/MODELS.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+| P0       | gemini-pro-vision | Not started | `IChatCompletion`      | Google API      | TBD          | TBD       | TBD      |
+| P0       | gemini-ultra      | Not started | `IChatCompletion`      | Google API      | TBD          | TBD       | TBD      |
+| P0       | embedding-001     | Not started | `IEmbeddingGeneration` | Google API      | TBD          | TBD       | TBD      |
```
Comment (Krzysztof318) [inline dotnet/docs/MODELS.md:64]: same as above

**Review by Krzysztof318 — CHANGES_REQUESTED** [review]
Update gemini model state

**Review by RogerBarreto — APPROVED** [review]

**Review by Krzysztof318 — APPROVED** [review]

**Comment by rosieks** [comment]
Could you please also document which models/deployments support function calling (and other features if exist). Not sure if that's the place or rather official documentation (ideally)

**Comment by Krzysztof318** [comment]
@rosieks 

Currently only openai supports Function calling in kernel. But soon gemini connector also will support this.

Most of connectors don't support (baked in) function calling. So I don't think we need to add this to the model doc.

File: dotnet/docs/MODELS.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+| P0       | text-embedding-ada-002  | Preview     | `IEmbeddingGeneration`         | Azure AI        | N/A          | N/A         | N/A      |
+| P0       | DALL·E 3                | Preview     | `ITextToImage`                 | OpenAI API      | N/A          | N/A         | N/A      |
+| P0       | DALL·E 3                | Preview     | `ITextToImage`                 | Azure AI        | N/A          | N/A         | N/A      |
+| P0       | Text-to-speech          | Complete    | `ITextToSpeech`                | OpenAI API      | TBD          | dmytrostruk | TBD      |
+| P0       | Speech-to-text          | Complete    | `ISpeechRecognition`           | OpenAI API      | TBD          | dmytrostruk | TBD      |
+| P1       | openai-whisper-large-v3 | Not started | `ISpeechRecognition`           | Azure AI        | TBD          | TBD         | TBD      |
+| P1       | openai-whisper-large-v3 | Not started | `ISpeechRecognition`           | Hugging Face    | TBD          | TBD         | TBD      |
+| P2       | Moderation              | Not started | `ITextClassification`          | OpenAI API      | TBD          | TBD         | TBD      |
```
Comment (Krzysztof318) [inline dotnet/docs/MODELS.md:42]:
```suggestion
| P2       | Moderation              | In Progress | `ITextClassification`          | OpenAI API      | TBD          | Krzysztof318 | MarkWallace |
```

Could you add small change?

**Review by SergeyMenshykh — APPROVED** [review]

**Pull request merged**